src/components/ParserComponent.tsx

```diff
@@ -121,6 +114,40 @@ export default class ParserComponent extends React.Component<ParserInterface,
       );
     }
 
+    /**
+     * Purpose: create a mock file event of the actual file for
+     * parse when it is recalled everytime the date format is changed
+     * @preconditions: a file with valid file data and
+     * a valid file name and a valid file type
+     * @return {any}: a mock file event of the actual file
+     */
+    createNewMockFile(): any {
+      const mockDateFile: File = new File(
+          [this.state.fileData],
+          String(this.state.fileName),
+          {type: this.props.fileType.mimeName},
+      );
+      // create file event of the mockfile and return it
+      return {target: {files: [mockDateFile]}};
+    }
+
+    /**
+   * Purpose: check if file is .csv when date format
+     * is changed and call parse if it is
+     * @preconditions: the current file should have a valid name (.csv)
+     * @postconditions: parse is called if file is .csv
+     * @return{boolean}: returns true if it works otherwise returns false
+   */
+    checkifCsvandcallParse(): boolean {
+      const nameOfFile = this.state.fileName;
+      const typeOfFile = nameOfFile.substr(nameOfFile.length - 4);
+      if (typeOfFile === '.csv') {
```

Comment:
Didn't we already do this in other places? And what about a .pdf that has its file ending changed to .csv?